realm/src/main/java/io/realm/Realm.java

```diff
@@ -1403,6 +1694,73 @@ public void executeTransaction(Transaction transaction) {
     }
 
     /**
+     * Similar to {@link #executeTransaction(Transaction)} but runs asynchronously from a worker thread
+     * @param transaction {@link io.realm.Realm.Transaction} to execute.
+     * @param callback optional, to receive the result of this query
+     * @return A {@link io.realm.Realm.Request} representing a cancellable task
+     */
+    public Request executeTransaction(final Transaction transaction, final Transaction.Callback callback) {
```

Comment:
I am not sure I like this override as it seems out of place compared to the rest of async naming. Wouldn't `executeTransactionAsync` fit better? Especially with the optional callback, in order to trigger async you would have to do:

```
realm.executeTransaction(new Realm.Transaction() { ... }, null);

// instead of 

realm.executeTransactionAsync(new Realm.Transaction() { .... });
realm.executeTransactionAsync(new Realm.Transaction() { .... }, new Realm.Transaction.Callback() { ... });
```